tasks/managed/collect-task-params/collect-task-params.yaml

```diff
@@ -0,0 +1,214 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: collect-task-params
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: |-
+    Tekton task that extracts values from a JSON file based on specified keys and exposes them as task results.
+    The task takes an array of resultIndex/key pairs and extracts the corresponding values from the dataPath JSON file.
+    Each extracted value is placed at the specified resultIndex in the extractedValues array for use by downstream
+    tasks. Optional default values can be provided for keys that may not exist in the data file. If no default is
+    provided and the key does not exist, the task will fail.
+  params:
+    - name: dataPath
+      type: string
+      description: Path to the JSON string of the merged data containing the values to extract
+    - name: keysToExtract
+      type: string
+      description: |
+        JSON array of objects with "resultIndex", "key", and optional "default" fields. Each object specifies the array
+        index where the extracted value should be placed, the JSON path key to extract from the data file, and an
+        optional default value to use if the key is not found. Example: [{"resultIndex": 1,
+        "key": ".releaseNotes.summary"}, {"resultIndex": 0, "key": ".foo", "default": "fallback_value"}]
+    - name: ociStorage
+      description: The OCI repository where the Trusted Artifacts are stored
+      type: string
+      default: "empty"
+    - name: ociArtifactExpiresAfter
+      description: Expiration date for the trusted artifacts created in the
+        OCI repository. An empty string means the artifacts do not expire
+      type: string
+      default: "1d"
+    - name: trustedArtifactsDebug
+      description: Flag to enable debug logging in trusted artifacts. Set to a non-empty string to enable
+      type: string
+      default: ""
+    - name: orasOptions
+      description: oras options to pass to Trusted Artifacts calls
+      type: string
+      default: ""
+    - name: sourceDataArtifact
+      type: string
+      description: Location of trusted artifacts to be used to populate data directory
+      default: ""
+    - name: dataDir
+      description: The location where data will be stored
+      type: string
+      default: $(workspaces.data.path)
+    - name: taskGitUrl
+      type: string
+      description: The url to the git repo where the release-service-catalog tasks and stepactions to be used are stored
+    - name: taskGitRevision
+      type: string
+      description: The revision in the taskGitUrl repo to be used
+  workspaces:
+    - name: data
+      description: The workspace where the dataPath json file resides
+  results:
+    - name: extractedValues
```

Comment:
Does this task  support extracting values that are more complex like objects and arrays, or just strings are supported?

Reply:
Just strings, but that is all we extract in `collect-*-params` tasks right now anyways. I don't think we use array or object results much at all in the catalog yet. I am not sure how this would work if a key was an object or array